### PR TITLE
Operator sdk v0.17.0 bump to use new flags

### PR DESCRIPTION
**Pull request by nikhil-thomas**
# Changes
Bumps `operator-sdk` version to `v0.17.0` 

The key reason for this bump is to make use of `--watch-namespace` flag in tests
Ref: https://github.com/operator-framework/operator-sdk/pull/2617

I shall update the Make file and Development doc in a separate pr.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
- update operator-sdk dependency to v0.17.0
```

### CODE REVIEW

**Comment by nikhil-thomas** [comment]
@timmyers

**Comment by vdemeester** [comment]
> {"component":"entrypoint","error":"could not start the process: exec: \"--scenario=kubernetes_execute_bazel\": executable file not found in $PATH","file":"prow/entrypoint/run.go:79","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","time":"2020-04-27T12:23:28Z"}

Oh that's my bade on `pull-tekton-operator-unit-tests`.. I missed something..

**Comment by vdemeester** [comment]
https://github.com/tektoncd/plumbing/pull/358 will fix that :angel:

**Comment by nikhil-thomas** [comment]
@sthaha

File: cmd/manager/main.go [hunk 1]
```diff
@@ -14,10 +14,14 @@ package main
...
 import (
 	"context"
+	"errors"
 	"flag"
 	"fmt"
+	"github.com/tektoncd/operator/version"
 	"os"
 	"runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"strings"
```
Comment (vdemeester) [inline cmd/manager/main.go:24]: This would benefit from `goimports` :stuck_out_tongue_closed_eyes:

File: cmd/manager/main.go [hunk 2]
```diff
@@ -125,7 +140,46 @@ func main() {
...
+	//// Add to the below struct any other metrics ports you want to expose.
+	//servicePorts := []v1.ServicePort{
+	//	{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+	//	{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
+	//}
+	//// Create Service object to expose the metrics port(s).
+	//_, err = metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	//if err != nil {
+	//	log.Info(err.Error())
+	//}
```
Comment (vdemeester) [inline cmd/manager/main.go:156]: Shouldn't this be removed ?

File: cmd/manager/main.go [hunk 3]
```diff
@@ -134,36 +188,70 @@ func main() {
...
+//	}
+//	// To generate metrics in other namespaces, add the values below.
+//	ns := []string{""}
+//	// Generate and serve custom resource specific metrics.
+//	err = kubemetrics.GenerateAndServeCRMetrics(cfg, ns, filteredGVK, metricsHost, operatorMetricsPort)
+//	if err != nil {
+//		return err
+//	}
+//	return nil
+//}
```
Comment (vdemeester) [inline cmd/manager/main.go:257]: Same here

File: cmd/manager/main.go (hunk 2 above)
Comment (nikhil-thomas) [reply cmd/manager/main.go:156]: ooops
removed

File: cmd/manager/main.go (hunk 1 above)
Comment (nikhil-thomas) [reply cmd/manager/main.go]: removed version for now. as we use version from a variable in config controller.

**Comment by nikhil-thomas** [comment]
@houshengbo 
could you take a look.
if we could merge this, i shall push update to the development doc especially with the new operator-sdk commands

File: cmd/manager/main.go [hunk 4]
```diff
@@ -14,10 +14,13 @@ package main
 
 import (
 	"context"
+	"errors"
 	"flag"
 	"fmt"
 	"os"
 	"runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"strings"
```
Comment (sthaha) [inline cmd/manager/main.go:23]: @nikhil-thomas this needs to change :( 
please try running go fmt

**Pull request merged**